File: CarbonCalcutaorBackEnd/src/main/java/net/atos/zerokhoi/entity/Actividad.java
package net.atos.zerokhoi.entity;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Table;
import javax.validation.constraints.NotEmpty;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.Data;
@Entity
@Table(name="actividades")
@Data
//@JsonIdentityInfo(generator=ObjectIdGenerators.IntSequenceGenerator.class, property="id")
public class Actividad implements Serializable{
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id;
	
	@Column(nullable= false)
	@NotEmpty
	private Date fecha;
	
	@Column(nullable= false)
	@NotEmpty
	private String categoria;
	
	@Column(nullable= false)
	@NotEmpty
	private Double consumo;
	
	
	@ManyToOne()
	@JoinColumn(name="user")	
	private Usuario user;

	
}

File: carbonCalculatorapp/src/app/vistas/bitacora/modelos/actividad.ts

export class Actividad {
    fecha: string;
    categoria: string;
    consumo: number;
    
    get _actividad(): string{
        return this.categoria;
    }

    set _actividad(actividad:string) {
        this.categoria = actividad;
    }

    get _consumo(): number{
        return this.consumo;
    }

    set _consumo(consumo:number) {
        this.consumo = consumo;
    }

    get _fecha(): string{
        return this.fecha;
    }

    set _fecha(fecha:string) {
        this.fecha = fecha;

    }
}